File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/dto/student/StudentExamResultResponse.java
package org.example.olympic_ot_project.dto.student;

import lombok.Builder;
import lombok.Data;
import org.example.olympic_ot_project.dto.exam.Leaderboard;

import java.util.List;

@Builder
@Data
public class StudentExamResultResponse {
    private Integer score;
    private Integer rank;
    private Integer totalParticipants;
    private List<Leaderboard> leaderboard;
}

File: frontend/olympic_it_project/olympic_it_project/lib/dto/profile/student_exam_result_response.dart
import 'package:olympic_it_project/dto/admin_manager/exam/leader_board_response.dart';

class StudentExamResultResponse {
  final int score;
  final int rank;
  final int totalParticipants;
  final List<LeaderBoardResponse> leaderboard;

  StudentExamResultResponse({
    required this.score,
    required this.rank,
    required this.totalParticipants,
    required this.leaderboard,
  });


  factory StudentExamResultResponse.fromJson(
      Map<String, dynamic> json,
  ) {

    return StudentExamResultResponse(

      score: json['score'] ?? 0,

      rank: json['rank'] ?? 0,

      totalParticipants:
          json['totalParticipants'] ?? 0,

      leaderboard:
          (json['leaderboard'] as List<dynamic>? ?? [])
              .map(
                (e) => LeaderBoardResponse.fromJson(
                  e as Map<String,dynamic>,
                ),
              )
              .toList(),

    );
  }
}